hbase-assembly/pom.xml

```diff
@@ -48,13 +48,17 @@
       <artifactId>hbase-shaded-mapreduce</artifactId>
     </dependency>
     <!-- Intra-project dependencies -->
+    <!-- We don't really add this to assembly tarball, we retain it here just to dump it into
```

Comment:
This looks iffy and fragile.
What about adding a new module which is used only for generating cached-classpath  instead ?
Or maybe using hbase-it, IF that already has all the dependencies we need for cached-classpath  ?

Reply:
I still don't like it, but I checked that we explictly list the dependencies to be included in the descriptors, so this is unlikely to cause regressions.

I'm going to work on the assemblies in the next sub-task anyway, I will check if can find a more elegant solution then.